ci/build.sh

```diff
@@ -0,0 +1,155 @@
+#!/bin/bash
```

Comment:
```shell
#!/usr/bin/env bash
```

Reply:
https://stackoverflow.com/questions/10376206/what-is-the-preferred-bash-shebang

Reply:
I think my way here is safer. /bin/bash must exist in MSYS2 and will not conflict with other places.

Reply:
Why would you want to limit this script to MSYS2? `mingw` runs on various different OSes and environments. This script new script also seems to deal with `MSVC`, which may be run without MSYS2, e.g. using WSL. Since Bash features aren't used (but certainly not inevitable) in this script, it'd be the best to use plain POSIX shell: `#!/bin/sh`

It'd be nice to run it through `shellcheck` and fix all the problems listed.

Reply:
It is only used for msys2 in appveyor. Use POSIX shell #!/bin/sh is ok too.